ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py

```diff
@@ -193,7 +204,18 @@ def _vrf_default_ipv4_static_routes(self) -> list[str]:
             for static_route in static_routes:
                 vrf_default_ipv4_static_routes.add(static_route["destination_address_prefix"])
 
-        return list(vrf_default_ipv4_static_routes)
+            vrf_default_redistribute_static = vrf_default.get("redistribute_static", vrf_default_redistribute_static)
```

Comment:
Looks like this logic could yield different results depending on how the tenants are named (and so ordered)

e.g.

# Example 1
```
tenant:
  - name: A
     vrfs:
       - name: default
         static_routes:
           - < some entry >
         redistribute_static: False
  - name: B
     vrfs:
       - name: default
          static_routes:
            - < some entry >
```

# Example 2

```
tenant:
  - name: A
     vrfs:
       - name: default
         static_routes:
           - < some entry >
  - name: B
     vrfs:
       - name: default
          static_routes:
            - < some entry >
          redistribute_static: False
```

# Rambling

Not sure what is the expected configuration for the config above to be honest .. so it may be a no problem or something to do with a route-map hehe..

This + delete line 196:

```suggestion
vrf_default_redistribute_static = vrf_default.get("redistribute_static", True)
```

Reply:
I changed it to respect a false set in any tenant, in the rare (and unsupported) case of having default in multiple tenants.
Future data model changes will remove this issue.

Reply:
Since I default to the previous setting, your two examples would render the same.